src/Pecee/Pixie/Exception.php

```diff
@@ -55,68 +55,62 @@ public static function create(PDOException $e, string $adapterName = null, Query
 
         switch ($adapterName) {
             case Mysql::class:
-                {
-                    // https://dev.mysql.com/doc/refman/5.6/en/error-messages-server.html
-                    if ((int)$errorSqlState === 23000) {
-                        switch ($errorCode) {
-                            case 1060: // Duplicate column name '%s'
-                                return new DuplicateColumnException($errorMsg, $errorCode, $e->getPrevious(), $query);
-                            case 1061: // Message: Duplicate key name '%s'
-                                return new DuplicateKeyException($errorMsg, $errorCode, $e->getPrevious(), $query);
-                            case 1062: // Message: Duplicate entry '%s' for key %d
-                                return new DuplicateEntryException($errorMsg, $errorCode, $e->getPrevious(), $query);
-                            case 1451: // Message: Cannot delete or update a parent row: a foreign key constraint fails (%s)
-                            case 1452: // Message: Cannot add or update a child row: a foreign key constraint fails (%s)
-                                return new ForeignKeyException($errorMsg, $errorCode, $e->getPrevious(), $query);
-                            case 1048: // Column '%s' cannot be null
-                                return new NotNullException($errorMsg, $errorCode, $e->getPrevious(), $query);
-                        }
-                    }
-
-                    if($exceptionErrorCode === 0) {
-                        break;
+                // https://dev.mysql.com/doc/refman/5.6/en/error-messages-server.html
+                if ((int)$errorSqlState === 23000) {
+                    switch ($errorCode) {
+                        case 1060: // Duplicate column name '%s'
```

Comment:
I have made a little mistake. Code 1060 and 1061 has different SQLSTATE than 23000.
I think, that this Exceptions could be removed because its are thrown during create tables not in typical usage (select/insert/update etc.)

I am sorry for my mistake
@mrcnpdlk

Reply:
Thanks for the heads up and no problem mate 👍 

We might be able to use it if we create a table scheme builder in the future. 

I added some unit tests for mysql - tested all error codes except the duplicatekey and foreign key exceptions. 

Great work! 😃